articles_controller.rb

```diff
@@ -91,4 +91,20 @@ def handle_forbidden_error
     render json: { errors: { article: ["not owned by user"] } },
            status: :forbidden
   end
+
+  def article_options
+    if params[:tag].present?
+      @articles.tagged_with(params[:tag])
+    elsif params[:author].present?
+      @articles.authored_by(params[:author])
+    elsif params[:favorited].present?
+      @articles.favorited_by(params[:favorited])
+    end
+  end
```

Comment:
Your code is really cool. You could change it to this just for the fun of doing things the ruby way.
```
def article_options
    @articles.tagged_with(params[:tag]) if params[:tag].present?
    @articles.authored_by(params[:author]) if params[:author].present?
    @articles.favorited_by(params[:favorited]) if params[:favorited].present?
    # as much as possible, also try to handle edge cases that don't fall within the if structure
  end
```
[link](https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals)

As a side note, as much as possible, give preference to immutability over mutablilty